File: Python/BOJ/스택의활용/3986.py
n = int(input())

result = 0
for _ in range(n):
  arr = list(input())
  
  stack = []
  for char in arr:
    if not stack or stack[-1] != char:
      stack.append(char)
    else:
      while stack and stack[-1] == char:
        stack.pop()

  if not stack:
    result += 1

print(result)

File: Swift/BOJ/2000/3986.swift
//
//  main.swift
//  SwiftAlgorithm
//
//  Created by 유정주 on 2022/03/03.
//

//3986 좋은 단어
import Foundation

let count = Int(readLine()!)!
var stack: [Character] = []
var result: Int = 0

for _ in 0..<count {
    stack = []
    let input = readLine()!
    
    for s in input {
        if let top = stack.last, top == s {
            stack.removeLast()
        } else {
            stack.append(s)
        }
    }
    
//    print("input: \(input) / stack: \(stack)")
    if stack.isEmpty {
        result += 1
    }
}

print(result)
